drivers/usb_c/tcpc/ucpd_numaker.c

```diff
@@ -0,0 +1,2558 @@
+{
+	return numaker_utcpd_bist_test_mode_set_enable(dev, enable);
+}
+
+/**
+ * @brief Sets the alert function that's called when an interrupt is triggered
+ *	  due to an alert bit
+ *
+ * @retval 0 on success
+ */
+static int numaker_tcpc_set_alert_handler_cb(const struct device *dev,
+					     tcpc_alert_handler_cb_t alert_handler,
+					     void *alert_data)
+{
+	struct numaker_tcpc_data *data = dev->data;
+
+	data->tcpc_alert.handler = alert_handler;
+	data->tcpc_alert.data = alert_data;
+
+	return 0;
+}
+
+/* Functions below with name pattern "*_tcpc_ppc_*" are to invoke by NuMaker PPC driver */
+
+int numaker_tcpc_ppc_is_dead_battery_mode(const struct device *dev)
+{
+	return numaker_utcpd_deadbattery_query_enable(dev);
+}
+
+int numaker_tcpc_ppc_exit_dead_battery_mode(const struct device *dev)
+{
+	return numaker_utcpd_deadbattery_set_enable(dev, false);
+}
+
+int numaker_tcpc_ppc_is_vbus_source(const struct device *dev)
+{
+	return numaker_utcpd_vbus_is_source(dev);
+}
+
+int numaker_tcpc_ppc_is_vbus_sink(const struct device *dev)
+{
+	return numaker_utcpd_vbus_is_sink(dev);
+}
+
+int numaker_tcpc_ppc_set_snk_ctrl(const struct device *dev, bool enable)
+{
+	const struct numaker_tcpc_config *const config = dev->config;
+	UTCPD_T *utcpd_base = config->utcpd_base;
+	uint32_t cmd;
+
+	if (enable) {
+		cmd = UTCPD_CMD_SINK_VBUS;
+	} else {
+		cmd = UTCPD_CMD_DISABLE_SINK_VBUS;
+	}
+	return NUMAKER_UTCPD_REG_WRITE_BY_NAME(dev, CMD, cmd);
+}
+
+int numaker_tcpc_ppc_set_src_ctrl(const struct device *dev, bool enable)
+{
+	const struct numaker_tcpc_config *const config = dev->config;
+	UTCPD_T *utcpd_base = config->utcpd_base;
+	uint32_t cmd;
+
+	if (enable) {
+		/* NOTE: Source VBUS high voltage (UTCPD_CMD_SRC_VBUS_NONDEFAULT) N/A */
+		cmd = UTCPD_CMD_SRC_VBUS_DEFAULT;
+	} else {
+		cmd = UTCPD_CMD_DISABLE_SRC_VBUS;
+	}
+	return NUMAKER_UTCPD_REG_WRITE_BY_NAME(dev, CMD, cmd);
+}
+
+int numaker_tcpc_ppc_set_vbus_discharge(const struct device *dev, bool enable)
+{
+	return numaker_utcpd_vbus_set_discharge(dev, enable);
+}
+
+int numaker_tcpc_ppc_is_vbus_present(const struct device *dev)
+{
+	return numaker_utcpd_vbus_is_present(dev);
+}
+
+int numaker_tcpc_ppc_set_event_handler(const struct device *dev, usbc_ppc_event_cb_t event_handler,
+				       void *event_data)
+{
+	struct numaker_tcpc_data *data = dev->data;
+
+	data->ppc_event.handler = event_handler;
+	data->ppc_event.data = event_data;
+
+	return 0;
+}
+
+int numaker_tcpc_ppc_dump_regs(const struct device *dev)
+{
+	return numaker_utcpd_dump_regs(dev);
+}
+
+/* End of "*_tcpc_ppc_*" functions */
+
+/* Functions below with name pattern "*_tcpc_vbus_*" are to invoke by NuMaker VBUS driver */
+
+bool numaker_tcpc_vbus_check_level(const struct device *dev, enum tc_vbus_level level)
+{
+	const struct numaker_tcpc_config *const config = dev->config;
+	UTCPD_T *utcpd_base = config->utcpd_base;
+	uint32_t mv_norm = 0;
```

Comment:
I think this does not need to be initialized

Reply:
yes, it needn't be initialized